algorithms/src/snark/marlin/marlin.rs

```diff
@@ -172,10 +174,10 @@ impl<E: PairingEngine, FS: AlgebraicSponge<E::Fq, 2>, MM: MarlinMode> MarlinSNAR
         end_timer!(sponge_time);
     }
 
-    fn absorb_with_msg(commitments: &[Commitment<E>], msg: &prover::ThirdMessage<E::Fr>, sponge: &mut FS) {
+    fn absorb_with_sums(commitments: &[Commitment<E>], sums: &[prover::MatrixSums<E::Fr>], sponge: &mut FS) {
```

Comment:
Why the change here?

Reply:
Because now we're not just sending sums in the matrix_sumcheck (now the fourth round), but also the lineval_sumcheck (now the third round). So I'm calling `absorb_labeled_with_sums` after both rounds.